rabbitsmpp/attributes.go

```diff
@@ -0,0 +1,71 @@
+package rabbitsmpp
+
+import (
+	"encoding/json"
+	"sync"
+)
+
+type Attributes struct {
+	mp    map[string]string
+	mutex sync.RWMutex
+}
+
+func NewAttributes() *Attributes {
```

Comment:
It'd be nice it to have another constructor which accepts a map as well. This way you'd only init it like we used to do not doing everything with `.Set` :

```
NewAttributes(map[string]string{
    "systemID":"test",
})
```

The important thing about implementing this is to copy the map (by iterating over it) so we have a safe copy on our side in case user decided to change his copy.